### ISSUE TITLE
Upgrade requiements

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/analysis-dashboard#192]
It would be good to upgrade to all the latest version, in particular pandas and numpy>=2
Upgrade geopandas too, which means we could get rid of the fiona dependency

We had some issues with streamlit dependencies, so it might be worth keeping them the same

https://github.com/Toblerity/Fiona/issues/1440#issuecomment-2328453074

**Comment by MAYANK12SHARMA** [comment]
Yes, @peterdudfield, I was also thinking of upgrading to Python 3.12 since some packages and functions have been deprecated. If you're on board with this, may I go ahead and proceed with upgrading to Python 3.12? 
I'm already using the updated version of Python with the new requirements, and it's working fine. Some packages, like `fiona` and `pygrib`, are causing issues with Python 3.11.

**Comment by peterdudfield** [comment]
please do
